tests/framework/microvm.py

```diff
@@ -750,60 +844,114 @@ def start(self, check=True):
             except KeyError:
                 assert self.started is True
 
+    def pause(self):
+        """Pauses the microVM"""
+        response = self.vm.patch(state="Paused")
+        assert self.api_session.is_status_no_content(response.status_code)
+
+    def resume(self):
+        """Resume the microVM"""
+        response = self.vm.patch(state="Resumed")
+        assert self.api_session.is_status_no_content(response.status_code)
+
     def pause_to_snapshot(
-        self, mem_file_path=None, snapshot_path=None, diff=False, version=None
+        self,
+        mem_file_path,
+        snapshot_path,
+        diff=False,
+        version=None,
     ):
         """Pauses the microVM, and creates snapshot.
 
         This function validates that the microVM pauses successfully and
         creates a snapshot.
         """
-        assert mem_file_path is not None, "Please specify mem_file_path."
-        assert snapshot_path is not None, "Please specify snapshot_path."
-
-        response = self.vm.patch(state="Paused")
-        assert self.api_session.is_status_no_content(response.status_code)
+        self.pause()
 
         response = self.snapshot.create(
-            mem_file_path=mem_file_path,
-            snapshot_path=snapshot_path,
+            mem_file_path=str(mem_file_path),
+            snapshot_path=str(snapshot_path),
             diff=diff,
             version=version,
         )
         assert self.api_session.is_status_no_content(
             response.status_code
         ), response.text
 
+    def make_snapshot(self, snapshot_type: str, target_version: str = None):
+        """Create a Snapshot object from a microvm."""
+        vmstate_path = "vmstate"
+        mem_path = "mem"
+        self.pause_to_snapshot(
+            mem_file_path=mem_path,
+            snapshot_path=vmstate_path,
+            diff=snapshot_type == "DIFF",
+            version=target_version,
+        )
+        root = Path(self.chroot())
+        return Snapshot(
+            vmstate=root / vmstate_path,
+            mem=root / mem_path,
+            disks=self.disks,
```

Comment:
"[test: revamp snapshot support to Microvm](https://github.com/firecracker-microvm/firecracker/pull/3896/commits/3197cde20ebb18fe9b6f96a66f43432d9760c06b)" depends on "[test: revamp .add_drive support](https://github.com/firecracker-microvm/firecracker/pull/3896/commits/5a13e878c0c196e51d6b4188fa69c589f978d2ac)"

Reply:
fixed it on my forked repo by reordering these two commits (the .add_drive commit comes first): https://github.com/zulinx86/firecracker/commits/new-ci-artifacts